### PR TITLE
add ad flags

### LINKED ISSUE #1341: Add advance settings for CLIs
### Description
Allowing `subql deployment:deploy` to include flags that enables advance settings, such as `unsafe`...etc.

### PR DESCRIPTION
**Pull request by bighammerw**
# Description
add advance setting flag for subql deployment :deployment 
Fixes #1341 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
## Checklist
- [ ] I have tested locally
- [ ] My code is up to date with the base branch
image
Screen Shot 2022-10-18 at 2 20 38 PM

### CODE REVIEW

File: packages/cli/src/commands/deployment/deploy.ts [hunk 1]
```diff
@@ -33,6 +33,18 @@ export default class Deploy extends Command {
...
+    batchSize: Flags.integer({description: 'Enter batchSize from 1 to 30', required: false}),
+    indexerSubscription: Flags.boolean({description: 'Enable Indexer subscription', required: false}),
+    historicalData: Flags.boolean({description: 'Enable Historical Data', required: false}),
+    workers: Flags.integer({description: 'Enter worker threads from 1 to 30', required: false}),
+    //query flags
+    queryUnsafe: Flags.boolean({description: 'Enable indexer unsafe', required: false}),
+    querySubscription: Flags.boolean({description: 'Enable Query subscription', required: false}),
+    queryTimeout: Flags.integer({description: 'Enter timeout from 1000ms to 60000ms', required: false}),
+    maxConnection: Flags.integer({description: 'Enter MaxConnection from 1 to 10', required: false}),
+    Aggregate: Flags.boolean({description: 'Enable Aggregate', required: false}),
```
Comment (stwiname) [inline packages/cli/src/commands/deployment/deploy.ts:47]: I think we should prefix all of these with `indexer` or `query`. We don't know what flags we will add in the future that could conflict

File: packages/cli/src/controller/deploy-controller.ts [hunk 2]
```diff
@@ -15,6 +15,8 @@ export async function deployToHostedService(
   endpoint: string,
   type: string,
   dictEndpoint: string,
+  query: object,
+  indexer: object,
```
Comment (stwiname) [inline packages/cli/src/controller/deploy-controller.ts:19]: We can define these types, better than object

File: packages/cli/src/commands/deployment/deploy.ts [hunk 3]
```diff
@@ -160,6 +191,8 @@ export default class Deploy extends Command {
       \nDictionary Endpoint: ${deploymentOutput.dictEndpoint}
       \nQuery URL: ${deploymentOutput.queryUrl}
       \nProject URL: ${BASE_PROJECT_URL}/project/${deploymentOutput.projectKey}
+      \nAdvance Setting of Indexer: ${deploymentOutput.configuration.config.indexer}
```
Comment (stwiname) [inline packages/cli/src/commands/deployment/deploy.ts:194]:
```suggestion
      \nAdvanced Settings for Indexer: ${deploymentOutput.configuration.config.indexer}
```

File: packages/cli/src/commands/deployment/deploy.ts [hunk 4]
```diff
@@ -160,6 +191,8 @@ export default class Deploy extends Command {
...
+      \nAdvance Setting of Query: ${deploymentOutput.configuration.config.query}
```
Comment (stwiname) [inline packages/cli/src/commands/deployment/deploy.ts:195]:
```suggestion
      \nAdvanced Settings for Query: ${deploymentOutput.configuration.config.query}
```

File: packages/cli/src/commands/deployment/deploy.ts [hunk 5]
```diff
@@ -68,6 +80,21 @@ export default class Deploy extends Command {
...
+      maxConnection: flags.maxConnection,
+      Aggregate: flags.Aggregate,
+    };
+    const indexerAD = {
+      unsafe: flags.unsafe,
+      batchSize: flags.batchSize,
+      subscription: flags.indexerSubscription,
+      historicalData: flags.historicalData,
+      workers: flags.workers,
+    };
```
Comment (stwiname) [inline packages/cli/src/commands/deployment/deploy.ts:96]: Do we need to do any validation on these? Or is that all handled on the hosted service? I see the descriptions say value ranges

Comment (bighammerw) [reply packages/cli/src/commands/deployment/deploy.ts:96]: I just tested it  in host service swagger, all the configurations will be verified on host service

Comment (bighammerw) [reply packages/cli/src/commands/deployment/deploy.ts:96]: Snip20221019_13

File: packages/cli/src/types.ts [hunk 6]
```diff
@@ -13,6 +13,21 @@ export interface ProjectSpecBase {
   endpoint: string;
 }
 
+export interface queryAD {
```
Comment (stwiname) [inline packages/cli/src/types.ts:16]: We can name this more clearly. Interfaces should also start with a capital

```suggestion
export interface QueryAdvancedOpts {
```

File: packages/cli/src/controller/deploy-controller.ts [hunk 7]
```diff
@@ -149,6 +151,8 @@ export async function redeploy(
   dictEndpoint: string,
   indexerVersion: string,
   queryVersion: string,
+  query: object,
+  indexer: object,
```
Comment (stwiname) [inline packages/cli/src/controller/deploy-controller.ts:155]: These types can be updated too

File: packages/cli/src/commands/deployment/deploy.ts [hunk 8]
```diff
@@ -33,6 +33,18 @@ export default class Deploy extends Command {
     queryVersion: Flags.string({description: 'Enter query-version', required: false}),
     dict: Flags.string({description: 'Enter dictionary', required: false}),
     endpoint: Flags.string({description: 'Enter endpoint', required: false}),
+    //indexer set up flags
+    indexer_unsafe: Flags.boolean({description: 'Enable indexer unsafe', required: false}),
```
Comment (stwiname) [inline packages/cli/src/commands/deployment/deploy.ts:37]: How are these flags used? Would it be `--query_batchSize`?

I think we can change these to camelCase like with queryVersion

Comment (bighammerw) [reply packages/cli/src/commands/deployment/deploy.ts:37]: yes will be like `--query_batchSize`,yup easy to change

**Review by stwiname — APPROVED** [review]

**Pull request merged**